### PR TITLE
Feature/validating webhook configuration

### PR DESCRIPTION
**Pull request by ycliuhw**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [ ] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [ ] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [ ] Do comments answer the question of why design decisions were made?

----

## Description of change

- Add `ValidatingWebhookConfigurations` support for k8s spec v2;
- Upgrade all `k8s.io` dependencies to k8s API version: `release-1.17;`

## QA steps

Deploy CaaS charm with below k8s spec;

```yaml
kubernetesResources:
  validatingWebhookConfigurations:
    pod-policy.example.com:
      - name: "pod-policy.example.com"
        rules:
          - apiGroups: [""]
            apiVersions: ["v1"]
            operations: ["CREATE"]
            resources: ["pods"]
            scope: "Namespaced"
        clientConfig:
          service:
            namespace: "example-namespace"
            name: "example-service"
          caBundle: "YXBwbGVz"
        admissionReviewVersions: ["v1", "v1beta1"]
        sideEffects: None
        timeoutSeconds: 5
  customResourceDefinitions:
    tfjobs.kubeflow.org:
      group: kubeflow.org
      scope: Namespaced
      names:
        kind: TFJob
        singular: tfjob
        plural: tfjobs
      versions:
        - name: v1
          served: true
          storage: true
      subresources:
        status: {}
      conversion:  # New fields.
        strategy: None
      validation:
        openAPIV3Schema:
          properties:
            spec:
              properties:
                tfReplicaSpecs:
                  properties:
                    # The validation works when the configuration contains
                    # `Worker`, `PS` or `Chief`. Otherwise it will not be validated.
                    Worker:
                      properties:
                        replicas:
                          type: integer
                          minimum: 1
                    PS:
                      properties:
                        replicas:
                          type: integer
                          minimum: 1
                    Chief:
                      properties:
                        replicas:
                          type: integer
                          minimum: 1
                          maximum: 1
```

```console
$ kubectl get ValidatingWebhookConfigurations t1-pod-policy.example.com -o yaml
apiVersion: admissionregistration.k8s.io/v1
kind: ValidatingWebhookConfiguration
metadata:
  annotations:
    juju.io/controller: d9ffeb33-ff69-482c-8a2c-ecd918496823
    juju.io/model: 654c5b0e-9baa-443b-8303-67ceb72a37d2
  creationTimestamp: "2020-02-13T06:09:14Z"
  generation: 1
  labels:
    juju-app: mariadb-k8s
    juju-model: t1
  name: t1-pod-policy.example.com
  resourceVersion: "2353335"
  selfLink: /apis/admissionregistration.k8s.io/v1/validatingwebhookconfigurations/t1-pod-policy.example.com
  uid: 3822dbab-c4f8-4f34-93d2-d0c8a8f5599f
webhooks:
- admissionReviewVersions:
  - v1
  - v1beta1
  clientConfig:
    caBundle: YXBwbGVz
    service:
      name: example-service
      namespace: example-namespace
      port: 443
  failurePolicy: Ignore
  matchPolicy: Exact
  name: pod-policy.example.com
  namespaceSelector: {}
  objectSelector: {}
  rules:
  - apiGroups:
    - ""
    apiVersions:
    - v1
    operations:
    - CREATE
    resources:
    - pods
    scope: Namespaced
  sideEffects: None
  timeoutSeconds: 5

$ kubectl get crd tfjobs.kubeflow.org -o jsonpath="{..conversion}"
map[strategy:None]

$ juju remove-application mariadb-k8s
removing application mariadb-k8s
- will detach storage database/2

$ kubectl get ValidatingWebhookConfigurations t1-pod-policy.example.com
Error from server (NotFound): validatingwebhookconfigurations.admissionregistration.k8s.io "t1-pod-policy.example.com" not found

$ kubectl get crd tfjobs.kubeflow.org
Error from server (NotFound): customresourcedefinitions.apiextensions.k8s.io "tfjobs.kubeflow.org" not found

```

## Documentation changes

Yes

## Bug reference

https://bugs.launchpad.net/juju/+bug/1859201
https://bugs.launchpad.net/juju/+bug/1862786

### CODE REVIEW

**Comment by ycliuhw** [comment]
[upgrade CRD support to non-beta](https://trello.com/c/HW876nmn/1046-upgrade-crd-support-to-non-beta)

**Comment by ycliuhw** [comment]
[k8s ValidatingWebhookConfiguration support](https://trello.com/c/VBhzgy26/1045-k8s-validatingwebhookconfiguration-support)

File: caas/kubernetes/provider/specs/v2_test.go [hunk 1]
```diff
@@ -539,6 +587,9 @@ password: shhhh`[1:],
 							Plural:   "tfjobs",
 							Singular: "tfjob",
 						},
+						Conversion: &apiextensionsv1beta1.CustomResourceConversion{
```
Comment (wallyworld) [inline caas/kubernetes/provider/specs/v2_test.go]: Is this the only new field? I thought there was another we could add to the test?

**Comment by ycliuhw** [comment]
> Will this still work on older k8s clusters with the beta apis?
> If not, we'll need to discuss the best approach....

So the new version we are upgrading to was just added a few new fields to the old v1beta1. 
The newer version v1 is not compatible with v1beta1.
Juju only supports one version of the API (which is still v1beta1 currently).
We need to get a solution to make k8s spec providing a way to specify the non-default API version to 
use(for example, v1) in the future.

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**